storybookjs/storybook
# Addon-docs: Lazy load Prettier in Vue and Angular

Issue:
In my exploration of the storybook bundle, I noticed that prettier adds a non-trivial amount to the bundle/ 

## What I did
I refactored the code to import prettier only when specifically needed.

The initial size of addon-controlsmdx--rounded from vue-kitchen-sink went to 1.8/7.54MB from 2.08/8.5MB (gzipped/original).

If docs.source.type is set to `CODE`, prettier will never load at all.

## How to test
in a Vue or Angular storybook, set docs.source.type is set to `'code'`. Use the network tab in your browser to track the total size of loaded JS. It should be lower than when the source type is not used.

I also tested by adding a delay to the async function where Prettier is loaded, and verified that the initial loading of the page is not blocked by Prettier being loaded. 

I don't think tests would be valuable for this.
I added an example of the Source block to vue-kitchen-sink
no


In the time when Prettier is being loaded, the original code of the story is shown. This isn't ideal, but I don't think it's a big issue since it should load relatively quick. If this is an issue, we can add a loading indicator to the source block, but this would require more refactoring.
